src/mame/video/model3.cpp

```diff
@@ -571,7 +532,7 @@ cached_texture *model3_state::get_texture(int page, int texx, int texy, int texw
 			return tex;
 
 	/* create a new texture */
-	tex = (cached_texture *)auto_alloc_array(machine(), uint8_t, sizeof(cached_texture) + (2 * pixwidth * 2 * pixheight) * sizeof(rgb_t));
+	tex = reinterpret_cast<cached_texture *>(new uint8_t[sizeof(cached_texture) + (2 * pixwidth * 2 * pixheight) * sizeof(rgb_t)]);
```

Comment:
This isn’t safe – `operator new` isn’t guaranteed to give suitably-aligned storage.

Also, is it guaranteed to clean up all of them on exit?  Previously, and that it didn’t explicitly free would get dumped with the pool, but now that safety net isn’t there.

Reply:
Good point about the alignment. I'll restructure a bit more aggressively so I can allocate `cached_texture` objects directly.

Reply:
See comment below about freeing.